### ISSUE TITLE
When to use bare symbols vs "string"

### ISSUE THREAD

**Issue by njtierney** [tidyverse/principles#101]
I'm curious what your thoughts are on when a user should be expected to use bare variables vs strings as arguments?

I couldn't find discussion on this so far, so apologies if this has already been covered - although I did see this phrase:

> Neither the `names_to` nor the `values_to` column exists in `relig_income`, so we provide them as character strings surrounded in quotes.

Stated around [this section](https://tidyr.tidyverse.org/articles/pivot.html#billboard) of the tidyr vignette.

So, my thoughts are this:

> Bare symbols should be used to reference something that already exists, or will exist, and not to denote options (except for boolean TRUE/FALSE).

For example:

  - an existing column in `select(data, barevar)`
  - A new column, such as in `mutate(data, newvar = barevar + 1)`

But not:

`my_plot(data, colour = red, line_type = dotted)`

**Comment by hadley** [comment]
Yeah, only use them via tidy evaluation, ie when they’re existing variables in some data frame.

**Comment by njtierney** [comment]
Ok great! Is this written down anywhere in the principles book?

**Comment by hadley** [comment]
Just here 😀

**Comment by njtierney** [comment]
Ok! I think this would be useful to document, can I suggest that this makes
it into the principles bookdown?
